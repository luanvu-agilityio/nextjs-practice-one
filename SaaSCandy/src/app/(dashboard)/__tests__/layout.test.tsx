import { render } from '@testing-library/react';
import DashboardLayout from '@/app/(dashboard)/layout';

describe('DashboardLayout - Snapshot Tests', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <DashboardLayout>
        <div>Test Content</div>
      </DashboardLayout>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render children correctly', () => {
    const { getByText } = render(
      <DashboardLayout>
        <div>Test Content</div>
      </DashboardLayout>
    );

    expect(getByText('Test Content')).toBeInTheDocument();
  });

  it('should wrap children in a div', () => {
    const { container } = render(
      <DashboardLayout>
        <div>Test Content</div>
      </DashboardLayout>
    );

    const wrapper = container.firstChild as HTMLElement;
    expect(wrapper.tagName).toBe('DIV');
  });
});
