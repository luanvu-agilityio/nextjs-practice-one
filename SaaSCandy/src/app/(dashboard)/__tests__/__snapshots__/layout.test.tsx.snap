// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DashboardLayout - Snapshot Tests should match snapshot 1`] = `
<div>
  <div>
    <div>
      Test Content
    </div>
  </div>
</div>
`;
